openapi.yml

```diff
@@ -0,0 +1,475 @@
+        - type: object
+          properties:
+            state:
+              $ref: '#/components/schemas/JobState'
+    OcrdExecutable:
+      type: string
+      pattern: '^ocrd-.*'
+    Processor:
+      description: The ocrd-tool.json for a specific tool
+      x-$ref: 'https://ocr-d.de/ocrd_tool.schema.json#/properties/tools/patternProperties/ocrd-.*'
+    ProcessorList:
+      description: List all available processors
+      type: array
+      items:
+        type: array
+        items:
+          $ref: '#/components/schemas/Processor'
+    ProcessorArgs:
+      description: The CLI arguments passed to an OCR-D processor
+      type: object
+      properties:
+        workspace: {$ref: '#/components/schemas/Workspace'}
+        input_file_grps: {type: string}
+        output_file_grps: {type: string}
+        page_id: {type: string}
+    ProcessorCall:
+      allOf:
+        - {$ref: '#/components/schemas/ProcessorArgs'}
+        - type: object
+          description: Full representation of a CLI call of a processor
+          required: ['executable']
+          properties:
+            executable: {$ref: '#/components/schemas/OcrdExecutable'}
+    ProcessorJob:
+      allOf:
+        - {$ref: '#/components/schemas/Job'}
+        - type: object
+          properties:
+            processor: {$ref: '#/components/schemas/Processor'}
+            workspace: {$ref: '#/components/schemas/Workspace'}
+    Workflow:
+      allOf:
+        - {$ref: '#/components/schemas/Resource'}
+    WorkflowJob:
+      allOf:
+        - {$ref: '#/components/schemas/Job'}
+        - type: object
+          properties:
+            workflow: {$ref: '#/components/schemas/Workflow'}
+            workspace: {$ref: '#/components/schemas/Workspace'}
+    DiscoveryResponse:
+      type: object
+      properties:
+        ram:
+          description: All available RAM in bytes
+          type: number
+        cpu_cores:
+          description: Number of available CPU cores
+          type: number
+        has_cuda:
+          description: Whether deployment supports NVIDIA's CUDA
+          type: boolean
+        cuda_version:
+          description: Major/minor version of CUDA
+          type: string
+        has_ocrd_all:
+          description: Whether deployment is based on ocrd_all
+          type: boolean
+        ocrd_all_version:
+          description: Git tag of the ocrd_all repo
+          type: string
+        has_docker:
+          description: Whether the OCR-D executables run in a docker container
+          type: boolean
+    LogEntry:
+      type: object
+      required:
+        - level
+        - message
+      properties:
+        level:
+          description: Log level
+          type: string
+          enum: ['debug', 'info', 'warning', 'error']
+        message:
+          description: Log message
+          type: string
+        time:
+          description: Log UTC time
+          type: string
```

Comment:
Maybe we could add a `pattern` here (and fry in regex hell).

Reply:
Better: add the format `date-time` which complies to https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14.

Reply:
```suggestion
          type: string
          format: date-time
```